engine/classes/Elgg/WidgetsService.php

```diff
@@ -108,17 +110,16 @@ public function createWidget($owner_guid, $handler, $context, $access_id = null)
 	 * @since 1.9.0
 	 */
 	public function canEditLayout($context, $user_guid = 0) {
-		$user = get_entity((int)$user_guid);
-		if (!$user) {
-			$user = _elgg_services()->session->getLoggedInUser();
+		try {
+			$user = _elgg_services()->entityTable->getUserForPermissionsCheck($user_guid);
+		} catch (UserFetchFailureException $e) {
+			return false;
 		}
 
-		$return = false;
-		if (_elgg_services()->session->isAdminLoggedIn()) {
```

Comment:
I assume this was a bug (checking if the logged in user was admin, even if you pass in a different GUID).